reconcile/terraform_cloudflare_dns.py

```diff
@@ -87,11 +92,34 @@ def run(self, dry_run: bool) -> None:
 
     @defer
     def run_with_defer(self, dry_run: bool, defer: Callable) -> None:
-        vault_settings = get_app_interface_vault_settings()
-        secret_reader = create_secret_reader(use_vault=vault_settings.vault)
+
+        settings = self._get_app_interface_settings()
+
+        if not settings.settings:
```

Comment:
Just a nit. I don't see the use `settings` alone, it's possible to instead of using `settings.settings[0]` we return `settings[0]` in `_get_app_interface_settings` and use here as `settings`